viewing_party/party.py

```diff
@@ -1,23 +1,152 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+    if not title or not genre or not rating:
+        return None
+    new_movie["title"] = title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+    return new_movie
+
+
+def add_to_watched(user_data, movie):
+    movies_watched_list = user_data["watched"]
+    movies_watched_list.append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    movies_to_watch_list = user_data["watchlist"]
+    movies_to_watch_list.append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    movies_to_watch_list = user_data["watchlist"]
+    movies_watched_list = user_data["watched"]
+
+    i = 0
+    for movie in movies_to_watch_list:
+        if movie["title"] == title:
+            movies_watched_list.append(movie)
+            del movies_to_watch_list[i]
+        i+=1
+    print(user_data)    
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    rating_sum = 0
+    rating_counter = 0
+    movies_watched_list = user_data["watched"]
+
+    for movie in movies_watched_list:
+        rating_sum += movie["rating"]
+        rating_counter += 1

```

Comment:
I noticed you are using a counter to keep track of the ratings but you could also track the ratings by using a list and grabbing the length of that list to calculate the sum. Something like this:
```
def get_watched_avg_rating(user_data):
    ratings = []
    for movie in movies_watched_list:
        ratings.append(movie["rating"])
    if not user_data["watched"]:
        return 0.0
    else:
        return (sum(ratings) / len(ratings))
```